File: archivist/src/test/java/com/zorroa/archivist/service/AnalystServiceTests.java
package com.zorroa.archivist.service;

import com.zorroa.archivist.AbstractTest;
import com.zorroa.common.domain.Analyst;
import com.zorroa.common.domain.AnalystBuilder;
import com.zorroa.sdk.domain.PagedList;
import com.zorroa.sdk.domain.Pager;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by chambers on 5/20/16.
 */
public class AnalystServiceTests extends AbstractTest {

    AnalystBuilder ping;

    @Before
    public void init() {
        ping = sendAnalystPing();
        refreshIndex();
    }

    @Test
    public void testGetAll() {
        PagedList<Analyst> list = analystService.getAll(Pager.first());
        assertEquals(1, list.size());
    }

    @Test
    public void testGet() {
        Analyst a = analystService.get(ping.getUrl());
        assertEquals(ping.getUrl(), a.getUrl());
    }

    @Test
    public void testGetById() {
        Analyst a1 = analystService.get(ping.getUrl());
        Analyst a2 = analystService.get(a1.getId());
        assertEquals(a1.getUrl(), a2.getUrl());
    }
}

File: archivist/src/test/kotlin/com/zorroa/archivist/service/AnalystServiceTests.kt
package com.zorroa.archivist.service

import com.zorroa.archivist.AbstractTest
import com.zorroa.archivist.security.AnalystAuthentication
import com.zorroa.common.domain.AnalystSpec
import org.junit.Before
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.context.SecurityContextHolder
import java.util.*
import kotlin.test.assertEquals

class AnalystServiceTests : AbstractTest() {

    @Autowired
    internal lateinit var analystService: AnalystService

    @Test
    fun testUpsert() {
        SecurityContextHolder.getContext().authentication = AnalystAuthentication("https://127.0.0.1:5000")
        val spec1 = AnalystSpec(
                1024,
                648,
                1024,
                0.5f,
                null)
        analystService.upsert(spec1)

        val spec2 = AnalystSpec(
                1024,
                1024,
                1024,
                1.0f,
                UUID.fromString("EF3B1E5A-31B5-4AEB-8C4E-7DA50F2AC592"))
        val a2 = analystService.upsert(spec2)
        assertEquals(spec2.totalRamMb, a2.totalRamMb)
        assertEquals(spec2.freeRamMb, a2.freeRamMb)
        assertEquals(spec2.load, a2.load)
    }
}